### ISSUE TITLE
Using GitHub to version cdktf projects' information

### ISSUE THREAD

**Issue by lsabreu96** [hashicorp/terraform-cdk#1478]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Hi, guys, not really sure if this is already implemented, but I couldn't find any guidance in the documentation.
Also, I'm not totally sure it makes, I'm a bit newbie to CDK and its best practices, but here it goes:

Basically I'm trying to use terraform-cdk with Github to version the `main.py` and the `cdktf.json` of different apps that will live in the same repository.
The problem appears mostly when collaborating on an existing project: if one updates its local branch, the person can't start the project inside the app, given it's not empty.

The workaround I thought so far is to update the local branch, create a temporary folder, start cdktf project in it, move all files inside this new folder and rename it to match the one existing inside GitHub.

For suggestion, I'd say to allow the init to start either on a non-empty folder or to consume some json to build the project up

**Comment by jsteinich** [comment]
A couple options come to mind:

1. Place the different apps in separate folders within the same repository.
2. Perhaps rather than separate apps, separate stacks within the same app better fits your situation.

**Comment by ansgarm** [comment]
Hi @lsabreu96!
What is the underlying workflow that you are trying to model? Do you want to create "preview environments" for people working locally? Why does `cdktf init` need to be run?

**Comment by lsabreu96** [comment]
Hi, @ansgarm !
The workflow would be a repository where each team can store the cdktf `main.py` file and any other related files, say source code for a lambda, for instance.

Not really preview environments for local working, more like I'm pulling changes from the remote and now I have a non empty folder from a project that I wish to work on, but I can't `cdktf init` because the repo isn't empty.

I thought that for every new project in every machine `cdktf init` was a requirement for the execution, but it seems that having `cdktf` installed and activating the virtual environment generated by Pip (I'm working with Python) seems to work just fine. Is this the correct path to follow here ?

**Comment by ansgarm** [comment]
> Is this the correct path to follow here ?

@lsabreu96 Yes, as @jsteinich already mentioned, `cdktf init` only creates the project. It differs from `terraform init` which needs to be run on each machine. The `cdktf-cli` takes care of running `terraform init` if needed (for example when running `cdktf plan`).

**Comment by ansgarm** [comment]
@lsabreu96
Additionally, if you want to create new projects based on existing ones, you don't need to use `cdktf init`, you could also copy the `cdktf.json` and other source files from an existing project into a new directory. You only need to make sure to rename a remote backend configuration (if used), i.e. change the Terraform Cloud Workspace name for the new project.
You could also create a custom [remote template](https://www.terraform.io/cdktf/create-and-deploy/remote-templates) that you could pass to `cdktf init` which could take care of setting up the files your projects need.

**Comment by lsabreu96** [comment]
Got it.
So it's a boilerplate, but a one that can be reproduced without the same command `cdktf init`. 
Thanks for the explanations, guys !

**Comment by ansgarm** [comment]
If anything else is left unclear, feel free to create a new issue or a post on [HashiCorp Discuss](https://discuss.hashicorp.com/c/terraform-core/cdk-for-terraform/47) and we're happy to help 👍
Can this issue be closed @lsabreu96?
